content/docs/faq-functions.md

```diff
@@ -177,3 +177,53 @@ class Alphabet extends React.Component {
   }
 }
 ```
+
+### How can I prevent a function from being called too quickly or too many times in a row?
+
+If you have an event handler such as `onClick` or `onScroll` and want to prevent the callback from being fired too quickly, you can wrap the handler with a utility such as [`_.debounce`](https://lodash.com/docs#debounce) or [`_.throttle`](https://lodash.com/docs#throttle). For a given `delay`, say `100ms`, debouncing calls the handler after activity stops for that amount of time; throttling prevents the handler from being called more than once per `delay`. See a visualization [here](http://demo.nimius.net/debounce_throttle/).
+
+Note: Call `event.persist()` when accessing events asynchronously to prevent the synthetic event from being recycled by [event pooling](/docs/events.html#event-pooling). See the debounce [example](#debounce) below.
+
+#### Throttle
+
+```jsx
+import throttle from 'lodash.throttle'
+
+class LoadMoreButton extends React.Component {
+  handleClick = throttle(() => {
+    this.props.loadMore()
+  }, 100)
+
+  render() {
+    return <button onClick={this.handleClick}>Load More</button>
+  }
+}
+```
+
+#### Debounce
+
+```jsx
+import debounce from 'lodash.debounce'
+
+class Searchbox extends React.Component {
+  handleChange = event => {
+    event.persist() // https://reactjs.org/docs/events.html#event-pooling
+    this._handleChangeDebounced(event.target.value)
```

Comment:
To clarify a bit: You only need to do 1 of the following:

1. Synchronously read the value from `event.target.value` and store it for later, or...
1. Call `event.persist()` so that React doesn't recycle the pooled event.

I think the former is better since we don't really need to freeze the whole event just to read that one value. I don't think we should show both because people are likely to assume you _need_ to do both and just copy-paste that pattern.

I would suggest changing this function to the something like:

```js
handleChange = event => {
  // React pools events, so we read the value before debounce.
  // Alternately we could call `event.persist()`
  // For more info see reactjs.org/docs/events.html#event-pooling
  this._handleChangeDebounced(event.target.value);
};
```

Reply:
Sounds good. At some point if there a lot of questions about it Event Pooling might need its own FAQ - "Why do am I getting `cannot read property value of null` in an event handler?"